### PR TITLE
add pagination to Search Page

### PR DESCRIPTION
**Pull request by fadykstas**

### CODE REVIEW

File: app/Actions/Place/GetPlaceCollectionByFilters/GetPlaceCollectionByFiltersAction.php [hunk 1]
```diff
@@ -28,6 +28,8 @@ public function execute(GetPlaceCollectionByFiltersRequest $request): GetPlaceCo
     {
         $categoryId = $request->getCategoryId();
         $location = $request->getLocation();
+        $page = $request->getPage() ? : 1;
```
Comment (nemoipaha) [inline app/Actions/Place/GetPlaceCollectionByFilters/GetPlaceCollectionByFiltersAction.php]: `?:` no extra page. 
We can add special const PAGE_DEFAULT = 1 into GetPlaceCollectionByFiltersRequest

File: app/Actions/Place/GetPlaceCollectionByFilters/GetPlaceCollectionByFiltersRequest.php [hunk 2]
```diff
@@ -8,7 +8,7 @@ class GetPlaceCollectionByFiltersRequest
     private $location;
     private $page;
 
-    public function __construct(int $page, ?int $category_id, ?string $location)
+    public function __construct(?int $page, ?int $category_id, ?string $location)
```
Comment (nemoipaha) [inline app/Actions/Place/GetPlaceCollectionByFilters/GetPlaceCollectionByFiltersRequest.php]: int $page is always present )

File: app/Actions/Place/GetPlaceCollectionByFilters/GetPlaceCollectionByFiltersRequest.php [hunk 3]
```diff
@@ -25,7 +25,7 @@ public function getLocation(): ?string
         return $this->location;
     }
 
-    public function getPage(): int
+    public function getPage(): ?int
```
Comment (nemoipaha) [inline app/Actions/Place/GetPlaceCollectionByFilters/GetPlaceCollectionByFiltersRequest.php]: : int

File: app/Actions/Place/GetPlaceCollectionByFilters/GetPlaceCollectionByFiltersRequest.php [hunk 4]
```diff
@@ -7,6 +7,7 @@ class GetPlaceCollectionByFiltersRequest
     private $category_id;
     private $location;
     private $page;
+    public const DEFAULT_PAGE = 1;
```
Comment (nemoipaha) [inline app/Actions/Place/GetPlaceCollectionByFilters/GetPlaceCollectionByFiltersRequest.php]: no need to add public )

File: app/Actions/Place/GetPlaceCollectionByFilters/GetPlaceCollectionByFiltersAction.php [hunk 5]
```diff
@@ -28,7 +28,7 @@ public function execute(GetPlaceCollectionByFiltersRequest $request): GetPlaceCo
     {
         $categoryId = $request->getCategoryId();
         $location = $request->getLocation();
-        $page = $request->getPage() ? : 1;
+        $page = $request->getPage() ? : $request::DEFAULT_PAGE;
```
Comment (nemoipaha) [inline app/Actions/Place/GetPlaceCollectionByFilters/GetPlaceCollectionByFiltersAction.php]: GetPlaceCollectionByFiltersRequest::DEFAULT_PAGE this is more explicit )

File: app/Actions/Place/GetPlaceCollectionByFilters/GetPlaceCollectionByFiltersRequest.php (hunk 3 above)
Comment (fadykstas) [reply app/Actions/Place/GetPlaceCollectionByFilters/GetPlaceCollectionByFiltersRequest.php]: Request can get `null` from Controller, so `: ?int`

**Pull request merged**